singer_sdk/plugin_base.py

```diff
@@ -162,10 +162,14 @@ def plugin_version(cls) -> str:
         Returns:
             The package version number.
         """
-        try:
-            version = metadata.version(cls.name)
-        except metadata.PackageNotFoundError:
-            version = "[could not be detected]"
+        distribution = get_project_distribution(__file__)
```

Comment:
tbf anything to do with `__file__` is a minefield (e.g. https://github.com/meltano/sdk/issues/875). It's problematic for standalone app distributions:

> ### `__file__` Ambiguity
> It isn’t clear whether `__file__` is actually required and what all is derived from existence of `__file__`. It also isn’t clear what `__file__` should be set to if it wouldn’t be a concrete filesystem path. Can `__file__` be virtual? Can it refer to a binary/archive containing the module?
> 
> Semantics of `__file__` need more clarification.

https://pyoxidizer.readthedocs.io/en/stable/pyoxidizer_technotes.html?#file-ambiguity

That's probably why it's failing on Windows.

Reply:
This was easy in the end - `cls` of the property references the current class (i.e. the subclass), so it was just a case of getting the module and its file `path = sys.modules[cls.__module__].__file__`. For SDK taps/targets, this will return `tap.py` or `target.py` 🎉